embedded/store/tx.go

```diff
@@ -25,31 +25,26 @@ import (
 )
 
 type Tx struct {
+	header *TxHeader
```

Comment:
Is there a reason why this is a pointer and not an instance?

Reply:
I also though about it, but left the pointer because it's exposing a pointer to the TxMetadata, which may be optional. So it won't be a safe plain copy. Do you think it should be anyways?